virtualizarr/writers/icechunk.py

```diff
@@ -124,15 +134,37 @@ def write_virtual_variable_to_icechunk(
     group: "Group",
     name: str,
     var: Variable,
+    append_dim: Optional[str] = None,
```

Comment:
Probably a down the road concern, but maybe we should add a validation / check that the append dim exists within the store.